src/ui/component/filePicker/filePicker.tsx

```diff
@@ -45,34 +44,67 @@ export const FilePicker: FC<FilePickerProps> = ({
     []
   )
 
-  const toFile = useCallback(
-    (item: DataTransferItem) => {
+  const traverseDirectory = useCallback(
+    async (fsDirectory: FileSystemDirectoryEntry): Promise<File[]> => {
+      const files: File[] = []
+      const directoryReader = fsDirectory.createReader()
+
+      const readDirEntries = async (): Promise<void> => {
+        const entries = await new Promise<FileSystemEntry[]>(resolve =>
+          directoryReader.readEntries(resolve)
+        )
+
+        for (const entry of entries) {
+          if (entry.isFile) {
+            const fsFileEntry = entry as FileSystemFileEntry
+            const file = await new Promise<globalThis.File>(resolve => fsFileEntry.file(resolve))
+            files.push(extractFileStream(file))
```

Comment:
If I understood correctly this part of code, the `File` interface extends from `Blob` which has a `stream()` property function which returns a ReadableStream :  `stream(): ReadableStream<Uint8Array>;`
So this function is used to extract the stream (as function is called) and to return a custom `File` type, which has the stream type set to `ReadableStream<Uint8Array>` instead of `() => ReadableStream<Uint8Array>`. 
Am I confusing something? Would you like me to change anything?

Reply:
@ccamel I suggest that we make the additional changes concerning the stream or the file url in another dedicated PR, so that we can already take advantage of this feature. Is that all right with you?

Reply:
@lolottetheclash Sure. No problem. :+1: